ckanext/datastore/backend/postgres.py

```diff
@@ -824,6 +824,8 @@ def convert(data: Any, type_name: str) -> Any:
     if type_name == 'nested':
         return json.loads(data[0])
     # array type
+    if type_name == 'numeric':
```

Comment:
there are a number of integer and floating point types in postgres that we could include here

Reply:
Another fix (suggested by @EricSoroos) is to expand the Python types check we do a few lines below to include `decimal`, which is how these are interpreted:

```python
   import decimal

   #...

    if isinstance(data, (int, float, decimal.Decimal)):
        return data
```

Just teste